core/src/main/scala/api/attributes.scala

```diff
@@ -35,15 +39,18 @@ object attributes {
   final class NodeType(`type`: String) extends api.ShowSelf {
     def to_s = `type`
   }
-  val File = new NodeType("file")
-  val Dir  = new NodeType("dir" )
-  val Link = new NodeType("link")
   implicit val _nodeType = new api.Key[NodeType]("type of node")
 
-  final case class Mtime(timestamp: Long)
+  final case class Mtime(timestamp: FileTime) extends FileTimeBased[Mtime](x => Mtime(x)) {
+    override def hashCode = timestamp.toMillis.##
+    override def equals(that: Any) = that match {
+      case Mtime(other) => timestamp.toMillis == other.toMillis
+      case _            => super.equals(that)
```

Comment:
Yeah, of course. What else could "equal" even mean? If FileTime has those methods implemented, what is the implementation? Because reference equality means those methods are not implemented, they're inherited from Object.

I'll probably throw out the java class, it isn't buying anything.